internal/ethapi/api.go

```diff
@@ -534,6 +534,49 @@ func (s *PublicBlockChainAPI) GetBalance(ctx context.Context, address common.Add
 	return b, state.Error()
 }
 
+// GetTransactionAndReceiptProof returns the Trie transaction and receipt proof of the given transaction hash.
+func (s *PublicBlockChainAPI) GetTransactionAndReceiptProof(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
+	tx, blockHash, _, index := core.GetTransaction(s.b.ChainDb(), hash)
+	if tx == nil {
+		return nil, nil
+	}
+	block, err := s.b.GetBlock(ctx, blockHash)
+	if err != nil {
+		return nil, err
+	}
+	tx_tr := deriveTrie(block.Transactions())
+
+	keybuf := new(bytes.Buffer)
+	rlp.Encode(keybuf, uint(index))
+	var tx_proof proofPairList
+	if err := tx_tr.Prove(keybuf.Bytes(), 0, &tx_proof); err != nil {
+		return nil, err
+	}
+	receipts, err := s.b.GetReceipts(ctx, blockHash)
+	if err != nil {
+		return nil, err
+	}
+	if len(receipts) <= int(index) {
+		return nil, nil
+	}
+	receipt_tr := deriveTrie(receipts)
+	var receipt_proof proofPairList
+	if err := receipt_tr.Prove(keybuf.Bytes(), 0, &receipt_proof); err != nil {
+		return nil, err
+	}
+	fields := map[string]interface{}{
```

Comment:
add stateProof

Reply:
hi, I thought you said just forget it... What is stateProof's input that you need? a transaction is not in the stateProof, only account balance, nonce etc are in stateProof

Reply:
This is only added for future use. If you need to use this interface directly, you don’t need to write it again.

Reply:
Got it. Can we do it in another PR?

Reply:
sure